### DISCUSSION TITLE
withRouter' is not exported from 'react-router-dom

### DISCUSSION THREAD

**Discussion by code-masala** [remix-run/react-router#8510] [General]
Is withRouter is replaced with other name?

**Comment by liuhanqu** [comment]
Which version ?

**Reply by code-masala** [reply]
6.1.1

**Reply by code-masala** [reply]
@liuhanqu  I saw many people facing this issue.But This is not mention in breaking changes

**Reply by code-masala** [reply]
Is there any update?

**Reply by liuhanqu** [reply]
The v6 have different api with v5.
I think you should read the doc first.

**Comment by code-masala** [comment]
Solution- create a function name as withRoute as it is removed .

```
import { useNavigate } from "react-router";
export const withRouter = (Component) => {
	const Wrapper = (props) => {
		const history = useNavigate();
		return <Component history={history} {...props} />;
	};
	return Wrapper;
};

```

**Reply by katonwd** [reply]
How can I use it?
`<WithRouter><Component /></WithRouter>` ?

**Comment by byteshiva** [comment]
What is the Equivalent of `withRouter` in v6?

**Reply by Vrq** [reply]
I assume that there is none, and you need to use the hooks to access navigation, etc.

**Comment by dev0T** [comment]
[First question](https://reactrouter.com/en/main/start/faq#what-happened-to-withrouter-i-need-it) from the FAQ
